client/src/components/TabbedPanel.vue

```diff
@@ -0,0 +1,69 @@
+<template>
+  <Fragment>
+    <TabList>
+      <Tab
+        v-for="tab in tabs"
+        :key="tab"
+        :class="[{ active: activeTab === tab }]"
+        @click="swithTab(tab)"
+      >
+        {{ tab }}
+      </Tab>
+    </TabList>
+    <component :is="currentTabComponent"></component>
+  </Fragment>
+</template>
+
+<script>
+import styled from 'vue-styled-components';
+import { Fragment } from 'vue-fragment';
+import QuestionTab from './QuestionTab.vue';
+
+const TabList = styled.ul`
+  margin-bottom: 15px;
+  padding: 0;
+  display: flex;
+  font-size: 1.2rem;
+  list-style-type: none;
+`;
+
+const Tab = styled.li`
+  padding: 8px 0;
+  margin-right: 40px;
+  cursor: pointer;
+
+  &.active {
+    font-weight: 600;
+    border-bottom: solid 1px grey;
+  }
+`;
+
+export default {
+  name: 'TabbedPanel',
+  data() {
+    return { activeTab: null };
+  },
+  props: {
+    tabs: Array,
+  },
+  components: {
+    TabList,
+    Tab,
+    Fragment,
+    questions: QuestionTab,
```

Comment:
That's a rough approach, cause we'll need to add any component that we might need as a tab here, and that isn't good at all. If there is a robust solution I'll take it